sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py

```diff
@@ -228,7 +228,18 @@ async def create_container(
         response_hook = kwargs.pop('response_hook', None)
         offer_throughput = kwargs.pop('offer_throughput', None)
         if offer_throughput is not None:
-            request_options["offerThroughput"] = offer_throughput
+            try:
+                if offer_throughput.auto_scale_max_throughput:
+                    request_options['autoUpgradePolicy'] = _stringify_auto_scale(offer=offer_throughput)
+                elif offer_throughput.auto_scale_increment_percent:
+                    raise ValueError("auto_scale_max_throughput must be supplied in "
+                                     "conjunction with auto_scale_increment_percent")
+                elif offer_throughput.offer_throughput:
+                    request_options["offerThroughput"] = offer_throughput.offer_throughput
+
+            except AttributeError:
+                if isinstance(offer_throughput, int):
+                    request_options["offerThroughput"] = offer_throughput
```

Comment:
Why is this logic being duplicated? It seems this logic should be refactored into one place.

Reply:
This logic is being duplicated because this is for the asynchronous version so we wanted to have the same code being used in the sync version here as well